pkg/collector/stats/types/types.go

```diff
@@ -38,90 +46,92 @@ func (s *UInt64Stat) SetNewDeltaValue(newDelta uint64, sum bool) error {
 		return nil
 	}
 	// verify overflow
-	if s.Aggr >= (math.MaxUint64 - newDelta) {
+	if s.aggr.Load() >= (math.MaxUint64 - newDelta) {
 		// we must set the the value to 0 when overflow, so that prometheus will handle it
-		s.Aggr = 0
+		s.aggr.Store(0)
 		return fmt.Errorf("the aggregated value has overflowed")
 	}
 	if sum {
 		// sum is used to accumulate metrics from different processes
-		s.Delta += newDelta
+		s.delta.Add(newDelta)
 	} else {
-		s.Delta = newDelta
+		if swapped := s.delta.CompareAndSwap(s.delta.Load(), newDelta); !swapped {
+			return fmt.Errorf("the delta value has been changed by another process")
```

Comment:
How about using just `Swap` ? 
```suggestion
		s.delta.Swap(newDelta)
```
Or use a loop to until it is swapped ? 
E.g. see: 

https://github.com/prometheus/client_golang/blob/bccd68204bf41e9cce27e8561b70e0d1cb350bcb/prometheus/counter.go#L137-L143